x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/signals/open_close_signals_route.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Hapi from 'hapi';
+import { DETECTION_ENGINE_SIGNALS_STATUS_URL } from '../../../../../common/constants';
+import { SignalsRequest } from '../../alerts/types';
+import { setSignalsStatusSchema } from '../schemas';
+import { ServerFacade } from '../../../../types';
+import { transformError } from '../utils';
+
+export const setSignalsStatusRouteDef: Hapi.ServerRoute = {
+  method: 'POST',
+  path: DETECTION_ENGINE_SIGNALS_STATUS_URL,
+  options: {
+    tags: ['access:siem'],
+    validate: {
+      options: {
+        abortEarly: false,
+      },
+      payload: setSignalsStatusSchema,
+    },
+  },
+  async handler(request: SignalsRequest, headers) {
+    const { signal_ids: signalIds, query, status } = request.payload;
+    const { callWithRequest } = request.server.plugins.elasticsearch.getCluster('data');
+    if (!callWithRequest) {
+      return headers.response().code(404);
```

Comment:
I don't do this check with my PR. I wonder if this is possible? We have checks like this with alerting and actions but really I think they are just always going to always be there and I'm wondering about this as well. It probably will always be there.

I think in other parts of the code no is checking for the existence of `elasticsearch`, they just rely on it being there and I am wondering if we should just do the same thing. Would cut down on unit testing something that is impossible and I could remove my unit tests around 404 for alerting/actions as well.

Reply:
I can remove it.